cli/flox/src/commands/activate.rs

```diff
@@ -671,6 +691,46 @@ impl Activate {
     }
 }
 
+/// Notify the user of available upgrades
+///
+/// Upon activation flox will start a detached process to check for upgrades.
+/// Future activations will be able to read the upgrade information from a file
+/// and notify the user if there are any upgrades available using this function.
+fn notify_upgrade_if_available(flox: &Flox, environment: &mut ConcreteEnvironment) -> Result<()> {
+    let upgrade_guard = UpgradeInformationGuard::read_in(environment.cache_path()?)?;
+
+    let Some(info) = upgrade_guard.info() else {
+        debug!("Not notifying user of upgrade, no upgrade information available");
+        return Ok(());
+    };
+
+    let current_lockfile = environment.lockfile(flox)?;
+
+    if Some(current_lockfile) != info.result.old_lockfile {
+        // todo: delete the info file?
+        debug!("Not notifying user of upgrade, lockfile has changed since last check");
+        return Ok(());
+    }
+
+    let diff = info.result.diff();
+    if diff.is_empty() {
+        debug!("Not notifying user of upgrade, no changes in lockfile");
+        return Ok(());
+    }
+
+    let description =
+        UninitializedEnvironment::from_concrete_environment(environment)?.message_description()?;
+
+    let message = formatdoc! {"
```

Comment:
If we decide to keep this notification on every single activate, I think it would be good to add a comment here with some justification. See https://flox-dev.slack.com/archives/C05P6A5J6U8/p1736465357236959

Reply:
I added some more background to the function description